samza-kafka/src/test/java/org/apache/samza/system/kafka/TestKafkaSystemConsumer.java

```diff
@@ -205,6 +215,156 @@ public void testFetchThresholdBytesDiabled() {
+
+    kafkaStartpointRegistrationHandler.visit(testSystemStreamPartition, startpointTimestamp);
+
+    // Mock verifications.
+    Mockito.verify(consumer).seek(testTopicPartition, Long.valueOf(testOffset));
+    Mockito.verify(consumer).offsetsForTimes(Mockito.anyMap());
+  }
+
+  @Test
+  public void testStartpointOldestVisitorShouldUpdateTheFetchOffsetInConsumer() {
+    // Define dummy variables for testing.
+    final Integer testPartitionId = 0;
+    final TopicPartition testTopicPartition = new TopicPartition(TEST_STREAM, testPartitionId);
+    final Partition testPartition = new Partition(testPartitionId);
+    final SystemStreamPartition testSystemStreamPartition = new SystemStreamPartition(TEST_SYSTEM, TEST_STREAM, testPartition);
+
+    final KafkaConsumer consumer = Mockito.mock(KafkaConsumer.class);
+    final KafkaStartpointRegistrationHandler
+        kafkaStartpointRegistrationHandler = new KafkaSystemConsumer.KafkaStartpointRegistrationHandler(consumer);
+
+    final StartpointOldest testStartpointSpecific = new StartpointOldest();
+
+    // Mock the consumer interactions.
+    Mockito.doNothing().when(consumer).seekToBeginning(ImmutableList.of(testTopicPartition));
+
+    // Invoke the consumer with startpoint.
+    kafkaStartpointRegistrationHandler.visit(testSystemStreamPartition, testStartpointSpecific);
+
+    // Mock verifications.
+    Mockito.verify(consumer).seekToBeginning(ImmutableList.of(testTopicPartition));
+  }
+
+  @Test
+  public void testStartpointUpcomingVisitorShouldUpdateTheFetchOffsetInConsumer() {
+    // Define dummy variables for testing.
+    final Integer testPartitionId = 0;
+    final TopicPartition testTopicPartition = new TopicPartition(TEST_STREAM, testPartitionId);
+    final Partition testPartition = new Partition(testPartitionId);
+    final SystemStreamPartition testSystemStreamPartition = new SystemStreamPartition(TEST_SYSTEM, TEST_STREAM, testPartition);
+
+    final KafkaConsumer consumer = Mockito.mock(KafkaConsumer.class);
+    final KafkaSystemConsumer.KafkaStartpointRegistrationHandler
+        kafkaStartpointRegistrationHandler = new KafkaSystemConsumer.KafkaStartpointRegistrationHandler(consumer);
+
+    final StartpointUpcoming testStartpointSpecific = new StartpointUpcoming();
+
+    // Mock the consumer interactions.
+    Mockito.doNothing().when(consumer).seekToEnd(ImmutableList.of(testTopicPartition));
+
+    // Invoke the consumer with startpoint.
+    kafkaStartpointRegistrationHandler.visit(testSystemStreamPartition, testStartpointSpecific);
+
+    // Mock verifications.
+    Mockito.verify(consumer).seekToEnd(ImmutableList.of(testTopicPartition));
+  }
+
+  @Test
+  public void testStartInvocationAfterStartPointsRegistrationShouldInvokeTheStartPointApplyMethod() {
+    // Initialize the constants required for the test.
+    Consumer mockConsumer = Mockito.mock(Consumer.class);
+    String testSystemName = "testSystem";
```

Comment:
Looks like some of these are already defined as static variables for the test class.